python/cudf/cudf/core/series.py

```diff
@@ -4786,6 +4787,11 @@ def cumprod(self, axis=0, skipna=True, *args, **kwargs):
                 )
                 result_col[first_index:] = None
 
+        if is_decimal_dtype(result_col.dtype):
```

Comment:
Can we move up this error before validating anything and also check for `self.dtype`?